### DISCUSSION TITLE
Sampler roadmap / overview

### DISCUSSION THREAD

**Discussion by MichaelClerx** [pints-team/pints#1603] [New methods]
Can't find an existing ticket for this, though sure we had one at some point.

Classed as black, _blue_, and **red** in Ben's diagram

Likelihood-free

- [ ] _Hamiltonian ABC_
- [ ] Markov chain monte-carlo ABC --> *TICKET OR PR NEEDED*
- [ ] _Regression-adjustment ABC_
- [x] Rejection ABC --> PR ~#881 #925~ #1442
- [x] **Sequential monte-carlo ABC** PR ~#1055~ #1442

Derivative-free

- [x] AM global-adaptation MCMC --> PR #491
- [x] AM local-componentwise MCMC --> PR #491
- [x] Delayed rejection adaptive metropolis (DRAM) --> PR #491
- [x] Differential evolution --> `DifferentialEvolutionMCMC`
- [x] Differential evolution adaptive metropolis (DREAM) --> `DreamMCMC`
- [x] Ellipsoidal nested sampling --> `NestedEllipsoidSampler`
- [x] Emcee hammer --> `EmceeHammerMCMC`
- [ ] Generalised elliptical slice sampling --> PR #900
- [ ] _Gradient-free HMC_ #101 
- [ ] Localised adaptive covariance --> PR #491
- [ ] _Mode jumping adaptive MCMC_ #492 
- [x] Population MCMC --> `PopulationMCMC`
- [x] Rao-blackwell AM MCMC --> PR #491
- [x] Random walk metropolis --> `MetropolisRandomWalkMCMC`
- [x] Remi AM MCMC --> `AdaptiveCovarianceMCMC`
- [x] Rejection nested sampling --> `NestedRejectionSampler`
- [ ] Sequential Monte Carlo --> PR #749
- [x] Slice sampling with doubling --> `SliceDoublingMCMC`
- [ ] Slice sampling with hyperrectangles --> PR #895
- [ ] Slice sampling with overrelaxation --> Temp PR #863
- [ ] _Slice sampling with Peano curves_ 
- [x] Slice sampling with stepout --> `SliceStepoutMCMC`
- [ ] _Transport map accelerated MCMC_
- [x] Vanilla AM MCMC --> PR #491
- [ ] _Vina copula adaptive MCMC_ Ticket #495 

1st order sensitivities

- [ ] _Adaptive HMC_
- [ ] _Adaptive MALA_ #adapti
- [ ] _Bouncy particle HMC_
- [ ] Covariance matching SS --> PR #896 
- [ ] _Diffusive nested sampling_ #152 
- [ ] _Dynamic nested sampling_ 
- [ ] _dyPolychord_
- [ ] _Galilean monte carlo_ #341 
- [x] Hamiltonian Monte Carlo (HMC) --> `HamiltonianMCMC`
- [x] Metropolis-adjusted langevin algorithm (MALA) --> `MALAMCMC`
- [x] Monomial gamma HMC --> `MonomialGammaHamiltonianMCMC`
- [ ] **Multinest** #1034 
- [ ] **No-U-Turn sampler (NUTS)** #1031 
- [ ] _Polychord_ #305 
- [x] Rank shrinking SS --> #899 
- [ ] _Shadow HMC_ #521 

2nd order sensitivities

- [ ] **Riemannian manifold HMC** 
- [ ] _Riemannian manifold MALA_

Framework for all sampling (temper distribution, sample from it, reweight via importance sampling):

- [ ] SIS MCMC --> PR #760

**Comment by MichaelClerx** [comment]
@ben could you check if the bottom 6 are e.g. different names for ones above? Found them on github but not in the pyramid

**Comment by MichaelClerx** [comment]
Aha, here it is: https://github.com/orgs/pints-team/projects/1#card-15881685

**Comment by ben18785** [comment]
Hi Michael, 

Thanks! I've edited the comment to remove those methods I think are duplicates. Not sure what to call SIS MCMC -- it's an approach that can be applied to all samplers (and likelihood-based optimisers) essentially. Just involves heating the distribution, sampling from it, then reweighting based on importance weights. Can leave it as it stands for now though!

**Comment by MichaelClerx** [comment]
Thanks!
